examples/example-phx-1_2/priv/repo/migrations/20180326205407_change_and_convert_page_data_type_in_backups_table.exs

```diff
@@ -0,0 +1,58 @@
+defmodule Example.Repo.Migrations.ChangeAndConvertPageDataTypeInBackupsTable do
+  @moduledoc false
+  use Ecto.Migration
+  alias Thesis.Backup
+  import Thesis.Config
+  import Ecto.Changeset, only: [change: 1, force_change: 3]
+
+  def change do
+    case convert_and_map_page_data() do
+      {:ok, backups} -> execute_migration(backups)
+      _ -> raise "Something went wrong with backup conversion. Please check \
+                  your thesis_backups table and re-run the migration."
+    end
+  end
+
+  def convert_and_map_page_data() do
+    backups = Enum.map(backups(), fn(b) ->
+      %{b | page_data: decompress_page_data(b.page_data)}
+    end)
+
+    {:ok, backups}
+  end
+
+  def execute_migration(backups) do
+    alter table(:thesis_backups) do
+      modify :page_data, :text
+    end
+
+    flush()
```

Comment:
We execute the migration before proceeding with saving the decompressed data back to this field